src/ParseQuery.js

```diff
@@ -1005,9 +1005,13 @@ class ParseQuery {
   *
   * @param {String} key The key that the string to match is stored in.
   * @param {String} value The string to search
+  * @param {Object} options (Optional) See https://docs.mongodb.com/manual/reference/operator/query/text/#text-query-operator-behavior
+  * @param {String} options.$language
+  * @param {String} options.$caseSensitive
+  * @param {String} options.$diacriticSensitive
   * @return {Parse.Query} Returns the query, so you can chain this call.
   */
- fullText(key: string, value: string): ParseQuery {
+ fullText(key: string, value: string, options: ?Object): ParseQuery {
```

Comment:
Can you add something similar to other functions that use options?

https://github.com/danielbm/Parse-SDK-JS/blob/4ea0c11bd375f0f1c8f942b7308e57ad9ba6ce30/src/ParseQuery.js#L495

Reply:
Done!

Reply:
I mean something like for every option.

```
var fullOptions = {};
    if (options.hasOwnProperty('language')) {
      fullOptions.$language = options.language;
    }
```

Can you also write a test case for $caseSensitive, $diaraticSensitivity, and options.unknown (this might throw an error)?

Reply:
Do you think this code would do the trick?

```javascript
    options = options || {};

    var fullOptions = {};
    for (var key in options) {
      switch (key) {
        case '$language':
          fullOptions.$language = options[key];
          break;
        case '$caseSensitive':
          fullOptions.$caseSensitive = options[key];
          break;
        case '$diacriticSensitive':
          fullOptions.$diacriticSensitive = options[key];
          break;
        default:
          throw new Error('Unknown option: ' + key);
          break;
      }
    }
```

Reply:
It should do the trick but seems verbose. Also I don't think we should have the `$` in the options getting passed in. $ is an internal thing. What do you think?

Reply:
@danielbm Can you write the test cases first?

Reply:
It seemed a bit verbose to me too, but doing the validation for the three options separately, plus the validation for unknown options was also. As per the ```$```, I totally agree. I'll remove it.

Reply:
Sweet!

Reply:
@dplewis I just pushed the code with the options validations, more test cases.